fus.c

```diff
@@ -0,0 +1,1002 @@
+          Solvable *s = pool_id2solvable (pool, *pp);
+          solvable_add_deparray (s, SOLVABLE_PROVIDES,
+                                 pool_str2id (pool, "module-default()", 1),
+                                 0);
+        }
+    }
+}
+
+static int
+repo_add_modulemd (Repo       *repo,
+                   FILE       *fp,
+                   const char *language,
+                   int         flags)
+{
+  g_autoptr(GPtrArray) objects = modulemd_objects_from_stream (fp, NULL);
+
+  _repo_add_modulemd_from_objects (repo, objects, language, flags);
+
+  return 0;
+}
+
+#ifdef FUS_TESTING
+static Repo *
+create_test_repo (Pool       *pool,
+                  const char *name,
+                  const char *type,
+                  const char *path)
+{
+  Repo *repo = repo_create (pool, name);
+
+  FILE *fp = fopen (path, "r");
+  /* Open a file with module metadata and load the content to the repo */
+  if (g_strcmp0 (type, "modular") == 0)
+    repo_add_modulemd (repo, fp, NULL, 0);
+  else
+    testcase_add_testtags (repo, fp, REPO_LOCALPOOL | REPO_EXTEND_SOLVABLES);
+
+  fclose (fp);
+
+  return repo;
+}
+
+#else
+
+static const char *
+repomd_find (Repo                 *repo,
+             const char           *what,
+             const unsigned char **chksump,
+             Id                   *chksumtypep)
+{
+  Pool *pool = repo->pool;
+  Dataiterator di;
+
+  dataiterator_init (&di, pool, repo, SOLVID_META, REPOSITORY_REPOMD_TYPE, what, SEARCH_STRING);
+  dataiterator_prepend_keyname (&di, REPOSITORY_REPOMD);
+
+  const char *filename = NULL;
+  *chksump = NULL;
+  *chksumtypep = 0;
+
+  if (dataiterator_step (&di))
+    {
+      dataiterator_setpos_parent (&di);
+      filename = pool_lookup_str (pool, SOLVID_POS, REPOSITORY_REPOMD_LOCATION);
+      *chksump = pool_lookup_bin_checksum (pool, SOLVID_POS, REPOSITORY_REPOMD_CHECKSUM, chksumtypep);
+    }
+
+  dataiterator_free (&di);
+
+  if (filename && !*chksumtypep)
+    {
+      g_warning ("No %s file checksum", what);
+      filename = NULL;
+    }
+
+  return filename;
+}
+
+static Repo *
+create_repo (Pool       *pool,
+             const char *name,
+             const char *path)
+{
+  FILE *fp;
+  Id chksumtype;
+  const unsigned char *chksum;
+  const char *fname;
+
+  Repo *repo = repo_create (pool, name);
+
+  fp = solv_xfopen (pool_tmpjoin (pool, path, "/", "repodata/repomd.xml"), "r");
+  repo_add_repomdxml (repo, fp, 0);
+  fclose (fp);
+
+  fname = repomd_find (repo, "primary", &chksum, &chksumtype);
+  if (fname)
+    {
+      fp = solv_xfopen (pool_tmpjoin (pool, path, "/", fname), 0);
+      repo_add_rpmmd (repo, fp, NULL, 0);
+      fclose (fp);
+    }
+
+  fname = repomd_find (repo, "group_gz", &chksum, &chksumtype);
+  if (!fname)
+    fname = repomd_find (repo, "group", &chksum, &chksumtype);
+  if (fname)
+    {
+      fp = solv_xfopen (pool_tmpjoin (pool, path, "/", fname), 0);
+      repo_add_comps (repo, fp, 0);
+      fclose (fp);
+    }
+
+#if 0
```

Comment:
isn't this old? I thought that we've fixed this. Did you rebase it properly?

Reply:
Let me recheck